### PR TITLE
Allow creating copies of `git_reference` objects.

### PR DESCRIPTION
**Pull request by arthurschreiber**
I'd like to make use of this in Rugged. 😄

### CODE REVIEW

File: tests/refs/dup.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{
+	g_repo = cl_git_sandbox_init("testrepo.git");
+}
+
+void test_refs_dup__cleanup(void)
+{
+	cl_git_sandbox_cleanup();
+}
+
+void test_refs_dup(void)
```
Comment (ethomson) [inline tests/refs/dup.c]: This test needs a proper name - `test_refs_dup` is the prefix for the `refs::dup` tests.  This test needs to be `test_refs_dup__direct` (for example).  (Clar is quite particular about naming.)

(You can run just the `refs::dup` tests to see that it only runs the one.)

File: tests/refs/dup.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{
+  git_reference *a, *b;
+
+	cl_git_pass(git_reference_lookup(&a, g_repo, "master"));
```
Comment (ethomson) [inline tests/refs/dup.c]: This should be `refs/heads/master` instead of `master`.

File: tests/refs/dup.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+	cl_assert(git_reference_cmp(a, b) == 0);
+
+	git_reference_free(b);
+	git_reference_free(a);
+}
+
+void test_refs_dup__symbolic(void)
+{
+  git_reference *a, *b;
```
Comment (ethomson) [inline tests/refs/dup.c]: Indentation is off here.

File: tests/refs/dup.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+void test_refs_dup__cleanup(void)
+{
+	cl_git_sandbox_cleanup();
+}
+
+void test_refs_dup(void)
+{
+  git_reference *a, *b;
```
Comment (ethomson) [inline tests/refs/dup.c]: The indentation here is a bit off.

**Comment by ethomson** [comment]
👍

**Comment by arthurschreiber** [comment]
Fixed everything up!

**Pull request merged**